feature 177624653 widget for single announcement

Ali-Amin opened the pull request:
**Story Description**
***Same as Webinar Widget***

As a user, I want to be able to open a widget for a single announcement where it has the fields provided in strapi's collection. 

- Not required fields (link  - deadline) should be displayed only if they are filled and should be hidden if they equal null.
- if the trivia field is empty, the button shouldn't appear. Otherwise, display a button that navigates me to the trivia form outside the app.

======

Note: This PR depends on Story `177624877 ` for some functionality

welo10 commented on lib/announcement_widget.dart:
```diff
@@ -0,0 +1,267 @@
+                  color: Colors.transparent,
+                  height: 10,
+                ),
+                SelectableText(
+                  'Date: ${DateFormat.yMMMMd('en_US').format(announcement.deadline)}',
+                  style: _style(context),
+                ),
```
The deadline field in the Announcements model is not required, so if any announcement without a deadline that woud raise an error as you expect always getting a deadline 
```suggestion
                (announcement.deadline != null)
                    ? SelectableText(
                        'Date: ${DateFormat.yMMMMd('en_US').format(announcement.deadline)}',
                        style: _style(context),
                      )
                    : Container(),

```

welo10 commented on lib/announcement_widget.dart:
```diff
@@ -0,0 +1,267 @@
+                  color: Colors.transparent,
+                  height: 10,
+                ),
+                SelectableText(
+                  'Time: ${DateFormat.Hm('en_US').format(announcement.deadline)}',
+                  style: _style(context),
+                ),
```
The deadline is already added above
```suggestion

```

Ali-Amin commented:
Addressed required changes and also added announcement descriptions to the widget.
####
![image](https://user-images.githubusercontent.com/41022464/115670360-9f91ba80-a349-11eb-9644-18b789eb5d8c.png)

welo10 commented:
Great work !
